### PR TITLE
updating top level structure of smidje to isolate parser and generator

### PR DESCRIPTION
**Pull request by ragnorak1000**
The parser files where moved into a parser directory
the parser no longer calls the generator
fact in parser was renamed to factparser and turned into a function
core has the fact macro which delegates to factparser and passes the result to the generator
added a data model doc

### CODE REVIEW

File: doc/data-model.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Data-Model
+
+smidje uses a two step process to generate tests from the provided midje syntax. It will first convert the test syntax
```
Comment (munk) [inline doc/data-model.md]: Nice documentation!

File: doc/data-model.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Data-Model
+
+smidje uses a two step process to generate tests from the provided midje syntax. It will first convert the test syntax 
+into a clojure map with all of the relevant test information. The map is then passed to a generator which will use it 
+to build out a cljs compatible test. Currently the only supported generator is the cljs-test generator which converts
+the map into cljs-test syntax.
+
+```clojure
+{:name-space ""
```
Comment (munk) [inline doc/data-model.md]: should this be :namespace ?

Comment (ragnorak1000) [inline doc/data-model.md]: yup updating

File: doc/data-model.md (hunk 1 above)
Comment (Luckvery) [inline doc/data-model.md]: +1

File: test/smidje/clj/cljs_generator/test_builder_test.clj [hunk 3]
```diff
@@ -1,6 +1,6 @@
-(ns smidje.cljs-generator.test-builder-test
+(ns smidje.clj.cljs-generator.test-builder-test
```
Comment (djljr) [inline test/smidje/clj/cljs_generator/test_builder_test.clj]: This ns name seems strange

**Pull request merged**